internal/gengapic/gengapic_test.go

```diff
@@ -378,6 +349,59 @@ methods:
 	}
 }
 
+func TestMethodDoc(t *testing.T) {
+	m := &descriptor.MethodDescriptorProto{
+		Name: proto.String("MyMethod"),
+	}
+
+	var g generator
+	g.comments = make(map[proto.Message]string)
+
+	for _, tst := range []struct {
+		in, want   string
+		deprecated bool
+	}{
+		{
+			in:   "",
+			want: "",
+		},
+		{
+			in:   "Does stuff.\n It also does other stuffs.",
+			want: "// MyMethod does stuff.\n// It also does other stuffs.\n",
+		},
+		{
+			in:         "Does not have a proper comment.",
+			want:       "// MyMethod does not have a proper comment.\n//\n// Deprecated: This may be removed in a future version.\n",
+			deprecated: true,
+		},
+		{
+			in:         "Deprecated: this is a proper deprecation notice.",
+			want:       "// Deprecated: this is a proper deprecation notice.\n",
```

Comment:
I think we still want the comment to start with the name of the thing being described (https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences)? So, it could be something like:

```
// Name is deprecated.
//
// Deprecated: this is a proper deprecation notice.
```

I could be wrong, but following the comment sentence convention every time seems good.

Reply:
Done.